### ISSUE TITLE
Prefer using `cache_key_with_version` over manual cache key

### ISSUE THREAD

**Issue by maestromac** [forem/forem#19102]
In the area where we use low-level caching, ie

```ruby
Rails.cache.fetch("article-cached-tagged-count-#{tag.name}", expires_in: 2.hours) do
  # anything goes
end
```

We should instead leverage Rail's method `cache_key_with_version`
```ruby
Rails.cache.fetch("#{tag.cache_key}/article-cached-tagged-count", expires_in: 2.hours) do
  # anything goes
end

# this will produce cache key similar to -> "tags/141-20230208011433457967/article-cached-tagged-count"
```

The reason is
1. It provides a more consistent and predictable style across all model specific cache
2. With the "version" automatically attached, it has the benefit of invalidating the cache whenever the model is updated.
3. This is what Rails's view caching uses internally

**Comment by Robeeerto** [comment]
@maestromac  I think I can do this.

**Comment by maestromac** [comment]
@Robeeerto yeah please go ahead. Please keep the PR small and target only one cache at a time. You may refer to [my first attempt](https://github.com/forem/forem/pull/19101) for inspiration. Thank you!

**Comment by Robeeerto** [comment]
@maestromac  Sure, I will slowly change part by part and send PR

**Comment by mirie** [comment]
As discussed in planning sync today, @maestromac will take a lookover at this issue / merged work and determine what remains
